src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -45,11 +48,26 @@ internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F,
             var param = F.Parameter(Parameters[0]);
             foreach (var field in ContainingType.GetFieldsToEmit())
             {
-                if (!field.IsStatic)
+                if (!field.IsStatic && field.ContainingType.Equals(ContainingType))
                 {
                     statements.Add(F.Assignment(F.Field(F.This(), field), F.Field(param, field)));
                 }
             }
         }
+
+        internal static MethodSymbol? FindCopyConstructor(NamedTypeSymbol containingType)
+        {
+            foreach (var member in containingType.InstanceConstructors)
+            {
+                if (member is MethodSymbol { IsStatic: false, ParameterCount: 1, Arity: 0 } method &&
+                    method.Parameters[0].Type.Equals(containingType, TypeCompareKind.CLRSignatureCompareOptions) &&
+                    method.Parameters[0].RefKind == RefKind.None)
+                {
+                    return method;
```

Comment:
>return method; [](start = 20, length = 14)

What about ambiguity cases? Given that we are using this function to check if constructor is a copy constructor, we are likely to get false negatives from the checks. Please add tests for scenarios like that.